inc/plugin-logic/class-admin-handler.php

```diff
@@ -206,16 +206,21 @@ public function expire_choctaw_events() {
 	private function get_event_expiry( array $time_and_date ): DateTime {
 		$timezone = new DateTimeZone( 'America/Chicago' );
 		if ( $time_and_date['is_all_day'] ) {
-			$expiry = new \DateTime( $time_and_date['start_date'], $timezone );
-			$expiry->modify( '+1 day' );
+			if ( $time_and_date['end_date'] ) {
+				$expiry = new \DateTime( $time_and_date['end_date'], $timezone );
+			} else {
+				$expiry = new \DateTime( $time_and_date['start_date'], $timezone );
+				$expiry->modify( '+1 day' );
+			}
 		} else {
 			if ( empty( $time_and_date['end_date'] ) ) {
 				$time_and_date['end_date'] = $time_and_date['start_date'];
 			}
 			if ( empty( $time_and_date['end_time'] ) ) {
-				$time_and_date['end_time'] = date( 'h:i A', strtotime( $time_and_date['start_time'] . ' +1 hour' ) ); // phpcs:ignore
+				$time_and_date['end_time'] = $time_and_date['start_time'];
```

Comment:
This will set the end_time to the start_time when it's not set regardless of other details.

So if the event goes from April 1 @ 8am - April 4, then the event will expire at April 4 @ 8am (when it should actually expire end of day on April 4).

I believe the logic should be, if no end_time is set, it should set the end_time to 11:59pm (unless there is also not an end_date, which is should then be set to start_time + 1 hour)

Reply:
resolved IRL